inc/class-wpseo-meta.php

```diff
@@ -1049,7 +1053,31 @@ public static function keyword_usage( $keyword, $post_id ) {
 			$post_ids = apply_filters( 'wpseo_posts_for_focus_keyword', $post_ids, $keyword, $post_id );
 		}
 
-		return $post_ids;
+		// Check if post ids is not empty.
+		if ( ! empty( $post_ids ) ) {
+			// Get the post subtypes for the posts that share the keyword.
+			$post_types = $repository->query()
+				->select( 'object_sub_type' )
+				->where_in( 'object_id', $post_ids )
+				->find_array();
+
+			// Get object_sub_type from each subarray in $post_ids.
+			$post_types = array_map(
+				function( $row ) {
+					return $row['object_sub_type'];
+				},
+				$post_types
+			);
+		}
+		else {
+			$post_types = [];
+		}
+
+
+		return [
```

Comment:
I think changing the return of a list of post ids to an array with multiple data structures can give some unexpected results in different places in the code. For example this function `keyword_usage` is used in the `class-post-metabox-formatter` file in a function called `get_keyword_usage_for_current_post` which in turn gets used by a function `get_focus_keyword_usage`. Here the result of the now changed data gets used as the basis of a filter. If we change this return and people (or ourselves in an add-on for example) expect the input of the filter to be an array of integers their code will probably fail.

I think all usages of the function you changed need to be checked and maybe refactored to take into consideration this new return value.

Reply:
Addressed this in the way we discussed by adding a new key to the object and a different routine for retrieving the data.